EngineAPI/src/main/java/nullengine/event/ComponentFilter.java

```diff
@@ -0,0 +1,30 @@
+package nullengine.event;
+
+import nullengine.component.Component;
+import nullengine.component.GameObject;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+public class ComponentFilter implements Predicate<GameObject> {
+    private Set<Class<? extends Component>> validation;
+
```

Comment:
If you want to specific game object (block, item) in event.
Use this helper class to filter the game object by component. (Or filter you self)

Not expect matching exactly class. I don't really expect dev should use OOP (BlockXX extends Block)
to make new block. 

Instead, they should register new block by grabbing different components. This solve the java OOP pain -- cannot have multi-parents.